### PR TITLE
[refactor](close) refactor ispendingfinish logic and close logic to do close more quickly

### PR DESCRIPTION
**Pull request by yiguolei**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/writer/vtablet_writer.h [hunk 1]
```diff
@@ -562,6 +558,8 @@ class VTabletWriter final : public AsyncResultWriter {
 
     Status _incremental_open_node_channel(const std::vector<TOlapTablePartition>& partitions);
 
+    void _do_try_close(RuntimeState* state, Status exec_status);
```
Comment (JackDrogon) [inline be/src/vec/sink/writer/vtablet_writer.h:561]: exec_status don't need to be passed by value

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/vec/sink/async_writer_sink.h [hunk 2]
```diff
@@ -113,7 +113,7 @@ class AsyncWriterSink : public DataSink {
         return Status::OK();
     }
 
-    bool is_close_done() override { return !_writer->is_pending_finish(); }
+    bool is_pending_finish() override { return _writer->is_pending_finish(); }
```
Comment (JackDrogon) [inline be/src/vec/sink/async_writer_sink.h:116]: mark as const & nodiscard

**Review by JackDrogon — CHANGES_REQUESTED** [review]

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**